storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3StorageConfig.java

```diff
@@ -209,12 +223,25 @@ public int uploadPartSize() {
         return getInt(S3_MULTIPART_UPLOAD_PART_SIZE_CONFIG);
     }
 
+    private URI s3ServiceEndpoint() {
+        final String url = getString(S3_ENDPOINT_URL_CONFIG);
+        if (url != null) {
+            try {
+                return new URI(url);
+            } catch (final URISyntaxException e) {
+                throw new ConfigException(S3_ENDPOINT_URL_CONFIG, url, "Must be a valid URI");
+            }
+        } else {
+            return null;
+        }
+    }
+
     private static class RegionValidator implements ConfigDef.Validator {
         @Override
         public void ensureValid(final String name, final Object value) {
             final String regionStr = (String) value;
             try {
-                Regions.fromName(regionStr);
+                Region.of(regionStr);
```

Comment:
I think this no longer works for validation, because `Region.of` doesn't seem to throw on an unknown region. OTOH, we can also accept everything the user provides as the region :man_shrugging: (i.e. drop the validator).

Reply:
Agree, dropping the validator since we should also allow custom regions